core/Plugin/Report.php

```diff
@@ -918,21 +918,42 @@ public function getThirdLeveltableDimension()
             return null;
         }
 
-        list($subtableReportModule, $subtableReportAction) = $this->getSubtableApiMethod();
+        return $this->getNthLeveltableDimension($level = 2);
+    }
 
-        $subtableReport = ReportsProvider::factory($subtableReportModule, $subtableReportAction);
-        if (empty($subtableReport) || empty($subtableReport->actionToLoadSubTables)) {
+    /**
+     * Returns the Dimension instance of the subtable report of this report's subtable report based on level.
+     *
+     * @param int $level The subTable level for which dimension is to be determined
+     * @return Dimension|null The subtable report's dimension or null if there is no subtable report or
+     *                        no dimension for the subtable report.
+     * @api
+     */
+    public function getNthLeveltableDimension(int $level): ?Dimension
+    {
+        if (empty($this->actionToLoadSubTables) || empty($level) || !is_numeric($level)) {
             return null;
         }
 
-        list($subSubtableReportModule, $subSubtableReportAction) = $subtableReport->getSubtableApiMethod();
+        if ($level === 1) {
+            return $this->getSubtableDimension();
+        }
 
-        $subSubtableReport = ReportsProvider::factory($subSubtableReportModule, $subSubtableReportAction);
-        if (empty($subSubtableReport)) {
+        [$subTableReportModule, $subTableReportAction] = $this->getSubtableApiMethod();
+        $subTableReport = ReportsProvider::factory($subTableReportModule, $subTableReportAction);
+        if (empty($subTableReport) || empty($subTableReport->actionToLoadSubTables)) {
             return null;
         }
+        for ($i = 2; $i <= $level; $i++) {
+            [$subSubTableReportModule, $subSubTableReportAction] = $subTableReport->getSubtableApiMethod();
+            $subSubTableReport = ReportsProvider::factory($subSubTableReportModule, $subSubTableReportAction);
+            if (empty($subSubTableReport)) {
+                return null;
+            }
+            $subTableReport = $subSubTableReport;
+        }
```

Comment:
Haven't tested below code, but there should be no need to handle the first level outside of the for loop.

```suggestion
        $subTableReport = $this;

        for ($i = 1; $i <= $level; $i++) {
            [$subSubTableReportModule, $subSubTableReportAction] = $subTableReport->getSubtableApiMethod();
            $subSubTableReport = ReportsProvider::factory($subSubTableReportModule, $subSubTableReportAction);
            if (empty($subSubTableReport) || empty($subTableReport->actionToLoadSubTables)) {
                return null;
            }
            $subTableReport = $subSubTableReport;
        }
```